### ISSUE TITLE
Should a light session be tolerant to failures?

### ISSUE THREAD

**Issue by cmeiklejohn** [dreyk/zraft_lib#33]
I'm reading and writing using a light_session -- the leader is node_1 and everything is zipping along fine -- reads and writes.  However, if node_1 is partitioned and then crashes, I see the leader transition to node_3, but writes using the light_session do not seem to complete?  When I debug the messages being sent, no matter which request I issue the read from, I always see the request go to node_1, which was that last known leader, who is no longer online.

The documentation seems to indicate that, with the light session configured to know about all peers, that a request would, if timed out, be rerouted to another node in the cluster.  Is this how it should work?

**Comment by ghost** [comment]
Seems like not a lot of issues get responded to in this repository (open since 2015) - maybe you’d be better off trying https://github.com/rabbitmq/ra <https://github.com/rabbitmq/ra> ?

**Comment by cmeiklejohn** [comment]
We're testing a variety of software as part of a research project, and this was one of the libraries we selected.

**Comment by dreyk** [comment]
@cmeiklejohn Yes. light_session should be tolerated to a node fails. It knows about all peers and should try to use other nodes to get a new leader.
